src/lib/pickles/impls.ml

```diff
@@ -6,14 +6,14 @@ module Wrap_impl = Snarky_backendless.Snark.Run.Make (Tock) (Unit)
 
 let test_bit x i = B.(shift_right x i land one = one)
 
-let forbidden_shifted_values ~modulus:r ~size_in_bits ~f =
+let forbidden_shifted_values ~modulus:r ~max_size ~size_in_bits ~f =
   let two_to_n = B.(pow (of_int 2) (of_int size_in_bits)) in
   let neg_two_to_n = B.(neg two_to_n) in
   let representatives x =
     let open Sequence in
     (* All values equivalent to x mod r that fit in [size_in_bits]
        many bits. *)
-    let fits_in_n_bits x = B.(x < two_to_n) in
+    let fits_in_n_bits x = B.(x < max_size) in
```

Comment:
another way to implement this could be to not have `~f` be part of that function and do the `List.map` on the caller site. Before applying `f` the caller can filter out anything that's bigger than the modulus if they want.

Reply:
so 

```ocaml
let modulus = Wrap_impl.Bigint.to_bignum_bigint Constant.size in
let size_in_bits = Constant.size_in_bits in
let values = forbidden_shifted_values ~size_in_bits ~modulus in
      values |> List.filter ~f:(fun x -> x < modulus) |> List.map ~f
```